cast/src/lib.rs

```diff
@@ -1656,6 +1661,75 @@ impl SimpleCast {
     pub fn disassemble(bytecode: &str) -> Result<String> {
         format_operations(disassemble_str(bytecode)?)
     }
+
+    /// Gets the selector for a given function signature
+    /// Optimizes if the `optimize` parameter is set to a number of leading zeroes
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use cast::SimpleCast as Cast;
+    ///
+    /// fn main() -> eyre::Result<()> {
+    ///     assert_eq!(Cast::get_selector("foo(address,uint256)")?.0, String::from("0xbd0d639f");
+    ///     assert_eq!(Cast::get_selector("foo(address,uint256)", 2)?.0, String::from("0x0000bf3e");
+    ///     assert_eq!(Cast::get_selector("foo(address,uint256)", 2)?.1, String::from("foo71661(address,uint256)");
+    ///
+    ///     Ok(())
+    /// }
+    /// ```    
+    pub fn get_selector(
+        signature: String,
+        optimize: Option<usize>,
+    ) -> Result<(String, Option<String>, Option<Duration>)> {
+        if optimize.is_none() || optimize.unwrap() > 4 {
+            let selector = HumanReadableParser::parse_function(&signature)?.short_signature();
+            return Ok((hex::encode(selector), None, None))
+        }
+        let mut name: &str = "";
+        let mut params: &str = "";
+        if let Some(index) = signature.find("(") {
+            name = &signature[..index];
+            params = &signature[index..];
+        }
+
+        let num_threads = num_cpus::get();
+        let start_time = Instant::now();
+        let found = AtomicBool::new(false);
+
+        let results: Vec<Option<(u32, String, String)>> = (0..num_threads)
+            .into_iter()
```

Comment:
this doesn't actually parallelize it, right?

I think we can use rayon's find_first here

https://docs.rs/rayon/latest/rayon/iter/trait.ParallelIterator.html#method.find_first

here

Reply:
ended up going with `.into_par_iter()` instead